### PR TITLE
[language][move] Add parsing tests [part 1]

### PR DESCRIPTION
**Pull request by bob-wilson**
This is the start of some tests for the parser, both positive and negative tests. I went through the current grammar and tried to construct examples with variations of various constructs, starting from the higher level things:

- File
- Module
- Address
- Use declarations
- Struct definitions
- Function declarations

Some of these things are likely to change as we refine the language but I went ahead and added tests for everything I could think of as it stands right now.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

I ran these new tests and they all pass

### CODE REVIEW

**Comment by bob-wilson** [comment]
I've updated this for the latest parser changes. It is ready to review now.

**Comment by bob-wilson** [comment]
I just realized that Todd recently moved some tests into a "parser" directory, where I had been using "parsing". I've changed this PR to match Todd's name.

File: language/move-lang/tests/move_check/parser/function_params_commas.move [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module M {
+    // Commas between function parameters are all optional.
+    f(p1: u64 p2: &u64) { }
```
Comment (tnowacki) [inline language/move-lang/tests/move_check/parser/function_params_commas.move]: Can we fix this... This is definitely not intentional. 

I'm fine with waiting for another PR

File: language/move-lang/tests/move_check/parser/function_type_trailing_comma.move [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module M {
+    fn<T,>() { } // Test a trailing comma in the type parameters
```
Comment (tnowacki) [inline language/move-lang/tests/move_check/parser/function_type_trailing_comma.move]: Oof, Didn't expect this to work. We should consider not allowing it

File: language/move-lang/tests/move_check/parser/module_struct_after_func.move [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module M {
+    fn() {}
+    // Struct declarations must come before function declarations.
```
Comment (tnowacki) [inline language/move-lang/tests/move_check/parser/module_struct_after_func.move]: We should fix this

File: language/move-lang/tests/move_check/parser/module_use_after_func.move [hunk 4]
```diff
@@ -0,0 +1,5 @@
+module M {
+    f() {}
+    // Use declarations must come before function declarations.
```
Comment (tnowacki) [inline language/move-lang/tests/move_check/parser/module_use_after_func.move]: This one will state for a bit (if not always)

I'm thinking of allowing use declarations at the beginning of any block `{ }`, wether it be for declaring an expression or a module

File: language/move-lang/tests/move_check/parser/struct_type_trailing_comma.move [hunk 5]
```diff
@@ -0,0 +1,3 @@
+module M {
+    struct S<T1, T2,> { } // Test a trailing comma in the type parameters
```
Comment (tnowacki) [inline language/move-lang/tests/move_check/parser/struct_type_trailing_comma.move]: Another one that we shouldn't allow

File: language/move-lang/tests/move_check/parser/function_params_commas.move (hunk 1 above)
Comment (bob-wilson) [reply language/move-lang/tests/move_check/parser/function_params_commas.move]: This is fixed as part of https://github.com/libra/libra/pull/1973

File: language/move-lang/tests/move_check/parser/function_type_trailing_comma.move (hunk 2 above)
Comment (bob-wilson) [reply language/move-lang/tests/move_check/parser/function_type_trailing_comma.move]: As discussed in https://github.com/libra/libra/pull/1973, we will continue to allow this.

File: language/move-lang/tests/move_check/parser/struct_type_trailing_comma.move (hunk 5 above)
Comment (bob-wilson) [reply language/move-lang/tests/move_check/parser/struct_type_trailing_comma.move]: As discussed in https://github.com/libra/libra/pull/1973, we will continue to allow this.

**Pull request closed without merging**